### ISSUE TITLE
output shape has a bug

### ISSUE THREAD

**Issue by QiyuLuo** [tefantasy/WeightedCorrelationExtension#2]
Thank you for your work, but I found a bug, when the input shapes H and W are different, **the output shape is not consistent with the original input shape**, which becomes H = W. 
```
import torch
from weighted_correlation import WeightedCorrelation
data2 = torch.randn(4, 64, 64, 128).cuda()
wc = WeightedCorrelation(64, 7, dilation=2).cuda()
output2 = wc(data2, data2)
print(f'data2 shape {data2.shape}')
print(f'output2 shape {output2.shape}')
>>
data2 shape torch.Size([4, 64, 64, 128])
output2 shape torch.Size([4, 49, 128, 128])
```
In addition, I have a question about whether the results of the correlation are normalized？
Looking forward to your early reply.

**Comment by tefantasy** [comment]
@QiyuLuo Hi, thanks for your comment.

Our (naive) implementation only supports inputs with H = W. This repo is just for reproducing results in the paper, so we do not have plan for extensions such as supporting inputs with different H and W. However any code contribution to this repo is welcome. 

The correlation results are normalized by the number of channels in a group. Using the annotations in the paper, this factor is g=C/G.

**Comment by QiyuLuo** [comment]
> @QiyuLuo Hi, thanks for your comment.
> 
> Our (naive) implementation only supports inputs with H = W. This repo is just for reproducing results in the paper, so we do not have plan for extensions such as supporting inputs with different H and W. However any code contribution to this repo is welcome.
> 
> The correlation results are normalized by the number of channels in a group. Using the annotations in the paper, this factor is g=C/G.

Thank you for your reply. It is a pity that I am not able to program CUDA at present and cannot complete its extension. I hope I can have the opportunity or someone else to implement it in the future .
